src/main/java/dev/jbang/GistFile.java

```diff
@@ -0,0 +1,60 @@
+package dev.jbang;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class GistFile {
+	public static boolean isGist;
```

Comment:
we shouldn't be keeping state that is specific to one file globally. this will definitely cause problems - especially during tests or when we have multiple gist references.

Reply:
I removed the `static`s that I added.
I needed to create an instance variable in `ScriptResource`:
```java
private List<Path> resolvedSourcePaths;
```
Basically if it's not `null` then there's no need to search sources.

And the good thing for gists is that it now handles `//SOURCES` in all files.
You can try this:
./jbang/bin/jbang https://gist.github.com/tivrfoa/8e6ea001f168fd4ef14763ceca3e5ab6

`one.java` references `two.java` that references `t3.java`

Reply:
is that only working for gist or you made it work for all resources ? 

that was on my todo list thus great if you made that work ;)

Reply:
Currently only gists (when starts with 'https://gist.github.com') are making use of that, but I think other resources will be able to use it too.